framework/src/multiapps/MultiApp.C

```diff
@@ -360,21 +366,27 @@ MultiApp::restore()
 }
 
 BoundingBox
-MultiApp::getBoundingBox(unsigned int app)
+MultiApp::getBoundingBox(unsigned int app, bool displaced_mesh)
 {
   if (!_has_an_app)
     mooseError("No app for ", name(), " on processor ", _orig_rank);
 
-  FEProblemBase & problem = appProblemBase(app);
+  unsigned int local_app = globalAppToLocal(app);
+  FEProblemBase & fe_problem_base = _apps[local_app]->getExecutioner()->feProblem();
+  MooseMesh & mesh = (displaced_mesh && fe_problem_base.getDisplacedProblem().get() != NULL)
+                         ? fe_problem_base.getDisplacedProblem()->mesh()
+                         : fe_problem_base.mesh();
 
   BoundingBox bbox = {};
   {
     Moose::ScopedCommSwapper swapper(_my_comm);
-    bbox = MeshTools::create_bounding_box(problem.mesh());
+    bbox = MeshTools::create_bounding_box(mesh);
```

Comment:
One issue here: you are doing this every time for each Transfer.  I wonder how long this takes... I think it's a complete iteration over all the nodes in the mesh....

Reply:
We usually store a bounding box on MooseMesh... can't we reuse that here?  (Maybe just get that one and inflate it slightly?)

Reply:
If the mesh moved, we have to get the up-to-date bounding box. That's my understanding of this. 

The only bounding box on `MooseMesh` I see is the one returned by `MooseMesh::getInflatedProcessorBoundingBox`. Am I looking at the wrong place?  Also, this one should give back the global one, correct?

I can throw a patch on top of this to re-compute the BBox only if `displaced_mesh` is true and pre-cache the BBox the first time we call this.

Reply:
Ah - you're right.

In that case - call `getInflatedBoundingBox()`... and cache the result when you can.